src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1496,13 +1493,10 @@ bool AbstractWriteLog<I>::check_allocation(
     }
     /* Don't attempt buffer allocate if we've exceeded the "full" threshold */
     if (m_bytes_allocated + bytes_allocated > m_bytes_allocated_cap) {
-      if (!req->has_io_waited_for_buffers()) {
-        req->set_io_waited_for_buffers(true);
-        ldout(m_image_ctx.cct, 5) << "Waiting for allocation cap (cap="
-                                  << m_bytes_allocated_cap
-                                  << ", allocated=" << m_bytes_allocated
-                                  << ") in write [" << *req << "]" << dendl;
-      }
+      ldout(m_image_ctx.cct, 20) << "Waiting for allocation cap (cap="
+                                 << m_bytes_allocated_cap
+                                 << ", allocated=" << m_bytes_allocated
+                                 << ") in write [" << *req << "]" << dendl;
```

Comment:
Can this result in these messages spamming the log?  It seems that previously `waited_buffers` served as a guard here and this message was printed only once per request.

Reply:
verified. Every `process_work()` will output once. Each time `alloc_resource()` failed will exit while(). Log are:
```
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 check_allocation: Waiting for allocation cap (cap=751619276, allocated=751566848) in write [image_extents=[[344457216,262144]], image_extents_summary=[total_bytes=262144, first_image_byte=344457216, last_image_byte=344719360], bl=buffer::list(len=262144,
  buffer::ptr(0~262144 0x55be6d902000 in raw 0x55be6d902000 len 262144 nref 1)
), user_req=0x55be6c230180, m_user_req_completed=0, m_deferred=1, detained=0]
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 wake_up: 
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 process_writeback_dirty_entries: Look for dirty entries
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 can_flush_entry:
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 process_writeback_dirty_entries: Next dirty entry isn't flushable yet
2021-12-29T16:26:40.417+0800 7fd5b5315700 10 librbd::cache::pwl::rwl::WriteLog: 0x55be63758b00 process_work: alloc_fail=1, allocated > high_water=1, allocated_entries > high_water=0
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::rwl::WriteLog: 0x55be63758b00 retire_entries: Look for entries to retire
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::AbstractWriteLog: 0x55be63758b00 can_retire_entry: 
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::rwl::WriteLog: 0x55be63758b00 retire_entries: Nothing to retire
2021-12-29T16:26:40.417+0800 7fd5b5315700 10 librbd::cache::pwl::rwl::WriteLog: 0x55be63758b00 process_work: Retired 0 times
2021-12-29T16:26:40.417+0800 7fd5b5315700 20 librbd::cache::pwl::rwl::WriteLog: 0x55be63758b00 alloc_resources:
```
After my modification https://github.com/ceph/ceph/pull/44157/commits/5573c234b1a001b40a992ba4e4eb017077ea08d7. It may reduce the output to the original similar conditions. Do you think such a change is necessary?

Reply:
If the log isn't spammed then probably not necessary.

Reply:
then drop it.